### PR TITLE
Add bamliquidator

### PR DESCRIPTION
**Pull request by JenkoB**
~~This tool needs docker image with new bamliquidator.~~ @dblenkus @tjanez

### CODE REVIEW

**Comment by mstajdohar** [comment]
Rebase

File: resolwe_bio/processes/plots/bamliquidator.yml [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+# =============
+---
+
+- slug: bamliquidator
+  name: Bamliquidator
+  requirements:
+    expression-engine: jinja
+    executor:
+      docker:
+        image: "resolwe/bio-linux8-resolwe-preview:bamliquidator"
```
Comment (dblenkus) [inline resolwe_bio/processes/plots/bamliquidator.yml]: Main docker image has been updated, so you can delete this.

File: resolwe_bio/processes/plots/bamliquidator.yml [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+      default: cell_type
+    - name: bin_size
+      label: Bin size
+      type: basic:integer
+      hidden: "analysis_type != 'bin'"
+      required: False
+      description: >
+        Number of base pairs in each bin. The smaller the bin size the longer the runtime and the larger the data files.
+        Default is 100000.
+    - name: regions_file_gtf
```
Comment (dblenkus) [inline resolwe_bio/processes/plots/bamliquidator.yml]: I think that `file` is redundant here, it would be better to use just `regions_gtf`

File: resolwe_bio/processes/plots/bamliquidator.yml [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+      label: Region gff file / Annotation file (.gff|.gtf)
+      type: data:annotation:gtf
+      hidden: "analysis_type != 'region'"
+      required: False
+    - name: regions_file_bed
```
Comment (dblenkus) [inline resolwe_bio/processes/plots/bamliquidator.yml]: Same here, `regions_bed`

File: resolwe_bio/processes/plots/bamplot.yml [hunk 4]
```diff
@@ -4,12 +4,12 @@
 ---
 
 - slug: bamplot
-  name: bamplot
+  name: Bamplot
```
Comment (dblenkus) [inline resolwe_bio/processes/plots/bamplot.yml]: Move bamplot fixes to separate commit.

File: resolwe_bio/processes/plots/bamplot.yml [hunk 5]
```diff
@@ -4,12 +4,12 @@
...
   requirements:
     expression-engine: jinja
   data_name: "bamplot"
-  version: 0.0.1
-  type: data:bam:plot
+  version: 1.0.1
```
Comment (dblenkus) [inline resolwe_bio/processes/plots/bamplot.yml]: Change this to `1.0.0`.

Comment (JenkoB) [inline resolwe_bio/processes/plots/bamplot.yml]: Wouldn't lowering version be a problem for platform?

File: resolwe_bio/processes/plots/bamliquidator.yml (hunk 2 above)
Comment (JenkoB) [inline resolwe_bio/processes/plots/bamliquidator.yml]: Resolwed

File: resolwe_bio/processes/plots/bamliquidator.yml [hunk 6]
```diff
@@ -0,0 +1,216 @@
...
+      hidden: "analysis_type != 'bin'"
+  run:
+    runtime: polyglot
+    language: bash
+    program: |
+
+
+      re-save analysis_type {{ analysis_type }}
+
+      {% if regions_gtf and regions_bed %}
```
Comment (dblenkus) [inline resolwe_bio/processes/plots/bamliquidator.yml]: Write all sanity checks in Django template language - last 3 are in bash. You should cover following cases:

- `{% if analysis_type == "bin" and (regions_bed or regions_gtf) %}`
- `{% if analysis_type == "bin" and not bin_size %}`
- `{% if analysis_type == "region" and (regions_bed or bin_size) %}`
- `{% if analysis_type == "region" and not regions_gtf %}`
- `{% if analysis_type == "bed" and (bin_size or regions_gtf) %}`
- `{% if analysis_type == "bed" and not regions_bed %}`

File: resolwe_bio/processes/plots/bamliquidator.yml [hunk 7]
```diff
@@ -0,0 +1,215 @@
...
+      ALL_MODE_PAR=()
+
+      BAM_PARAMS+=({% for bam_file in bam %}"{{bam_file.bam.file}}" {% endfor %})
+      BAM_BAI_PARAMS+=({% for bai_file in bam %}"{{bai_file.bai.file}}" {% endfor %})
+      SKIP_PLOT='{% if skip_plot %}--skip_plot {% endif %}'
+      BLACK_LIST='{% if black_list %}--black_list {{black_list|join(' ')}} {% endif %}'
+      SENSE='{% if sense %}--sense {{ sense }} {% endif %}'
+      EXTENSION='{% if extension %}-e {{ extension }} {% endif %}'
+
+      ALL_MODE_PAR+=("{{ cell_type }}" -n {{threads}} --xml_timings ${SKIP_PLOT} ${BLACK_LIST} ${EXTENSION} ${SENSE} -o output)
```
Comment (dblenkus) [inline resolwe_bio/processes/plots/bamliquidator.yml]: Change this to string.

Does it make sense to include `SKIP_PLOT` for all modes if it is also relevant for `bin` mode?

File: resolwe_bio/processes/plots/bamliquidator.yml [hunk 8]
```diff
@@ -0,0 +1,215 @@
...
+        {% else %}
+          BIN_SIZE=10000
+        {% endif %}
+
+        bamliquidator_batch.py "${ALL_MODE_PAR[@]}" -b ${BIN_SIZE} -f
+        re-checkrc "Bamliquidator failed."
+
+        {% if not skip_plot %}
+          SUMMARY_URL=()
+          SUMMARY_URL+=("{\"file\":\"output/summary.html\"}")
```
Comment (dblenkus) [inline resolwe_bio/processes/plots/bamliquidator.yml]: Is there a reason why is this a list? Probably you can just change field type to `basic:file:html`.

File: resolwe_bio/processes/plots/bamliquidator.yml (hunk 6 above)
Comment (JenkoB) [inline resolwe_bio/processes/plots/bamliquidator.yml]: {% if analysis_type == "bin" and not bin_size %} is not needed, since default bin size is set.

Is any basic difference if is written in bash or in Django?

File: resolwe_bio/processes/plots/bamliquidator.yml (hunk 7 above)
Comment (JenkoB) [inline resolwe_bio/processes/plots/bamliquidator.yml]: I have added SKIP_PLOT to bin mode, thanks.
What is wrong with array?

File: resolwe_bio/processes/plots/bamliquidator.yml (hunk 8 above)
Comment (JenkoB) [inline resolwe_bio/processes/plots/bamliquidator.yml]: If it is not list I got Error:
```TypeError: list indices must be integers or slices, not str```

Comment (dblenkus) [inline resolwe_bio/processes/plots/bamliquidator.yml]: Probably bacause you have to use `re-save-file` instead of `re-save`.

File: resolwe_bio/processes/plots/bamliquidator.yml (hunk 6 above)
Comment (dblenkus) [inline resolwe_bio/processes/plots/bamliquidator.yml]: No, you can use any of them, just don't mix them to keep code more consistent.

**Comment by mstajdohar** [comment]
Please rebase again
